pkg/shp/shapefile_test.go

```diff
@@ -0,0 +1,197 @@
+package shp
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBox_ExtendWithPoint(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name  string
+		input struct {
+			b Box
+			p Point
+		}
+		expected Box
+	}{
+		{
+			name: "MaxY",
+			input: struct {
+				b Box
+				p Point
+			}{
+				b: Box{0, 0, 1, 1},
+				p: Point{0, 2}},
+			expected: Box{
+				MinX: 0,
+				MinY: 0,
+				MaxX: 1,
+				MaxY: 2,
+			},
+		},
+		{
+			name: "MaxX",
+			input: struct {
+				b Box
+				p Point
+			}{
+				b: Box{0, 0, 1, 1},
+				p: Point{2, 0}},
+			expected: Box{
+				MinX: 0,
+				MinY: 0,
+				MaxX: 2,
+				MaxY: 1,
+			},
+		},
+		{
+			name: "MinX",
+			input: struct {
+				b Box
+				p Point
+			}{
+				b: Box{0, 0, 1, 1},
+				p: Point{-1, 0}},
+			expected: Box{
+				MinX: -1,
+				MinY: 0,
+				MaxX: 1,
+				MaxY: 1,
+			},
+		},
+		{
+			name: "MinY",
+			input: struct {
+				b Box
+				p Point
+			}{
+				b: Box{0, 0, 1, 1},
+				p: Point{0, -1}},
+			expected: Box{
+				MinX: 0,
+				MinY: -1,
+				MaxX: 1,
+				MaxY: 1,
+			},
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(tt *testing.T) {
+			tt.Parallel()
+			tc.input.b.ExtendWithPoint(tc.input.p)
+			assert.Equal(tt, tc.expected, tc.input.b)
+		})
+	}
+
+}
+
+func TestBox_Extend(t *testing.T) {
+	a := Box{-124.763068, 45.543541, -116.915989, 49.002494}
+	b := Box{-92.888114, 42.49192, -86.805415, 47.080621}
+	a.Extend(b)
+	c := Box{-124.763068, 42.49192, -86.805415, 49.002494}
+	if a.MinX != c.MinX {
+		t.Errorf("a.MinX = %v, want %v", a.MinX, c.MinX)
+	}
+	if a.MinY != c.MinY {
+		t.Errorf("a.MinY = %v, want %v", a.MinY, c.MinY)
+	}
+	if a.MaxX != c.MaxX {
+		t.Errorf("a.MaxX = %v, want %v", a.MaxX, c.MaxX)
+	}
+	if a.MaxY != c.MaxY {
+		t.Errorf("a.MaxY = %v, want %v", a.MaxY, c.MaxY)
+	}
+}
+
+func TestNewPolyLine(t *testing.T) {
+
```

Comment:
```suggestion
```

The first blank line of the function is included without any regularity. It would be better to either erase them all or include them all.